crates/wash/src/cli/update.rs

```diff
@@ -67,25 +68,99 @@ pub struct UpdateCommand {
     #[clap(long, short = 'd')]
     dry_run: bool,
 
-    // TODO(#30): support --major, --minor, --patch flags to control update granularity
     /// Point at a different repository for updates
     #[clap(long, default_value = REPO)]
     git: String,
 
     /// GitHub token for private repository access. Can also be set via GITHUB_TOKEN, GH_TOKEN, or GITHUB_ACCESS_TOKEN environment variables
     #[clap(long, env = "WASH_GITHUB_TOKEN")]
     token: Option<String>,
+
+    /// Allow major version updates (breaking changes)
+    #[clap(long, conflicts_with_all = ["minor", "patch"])]
+    major: bool,
+
+    /// Allow minor version updates (new features, no breaking changes)
+    #[clap(long, conflicts_with_all = ["major", "patch"])]
+    minor: bool,
+
+    /// Allow only patch updates (bug fixes only)
+    #[clap(long, conflicts_with_all = ["major", "minor"])]
+    patch: bool,
+}
+
+fn parse_version(tag: &str) -> Option<Version> {
+    let t = tag.strip_prefix("wash-").unwrap_or(tag);
+    let t = t.strip_prefix('v').unwrap_or(t);
+
+    match Version::parse(t) {
+        Ok(version) => {
+            trace!("Parsed version '{}' as {}", tag, version);
+            Some(version)
+        }
+        Err(e) => {
+            trace!("Failed to parse version '{}': {}", tag, e);
```

Comment:
supernit preference, just a code style nit that I try to keep consistent throughout the repo!
```suggestion
            trace!("Parsed version '{tag}' as {version}");
            Some(version)
        }
        Err(e) => {
            trace!("Failed to parse version '{tag}': {e}");
```